src/manager.cpp

```diff
@@ -210,12 +214,47 @@ types::DbusVariantType
     Manager::readKeyword(const types::Path i_fruPath,
                          const types::ReadVpdParams i_paramsToReadData)
 {
-    // Code to supress unused variable warning. To be removed.
-    (void)i_fruPath;
-    (void)i_paramsToReadData;
+    try
+    {
+        nlohmann::json l_jsonObj{};
+
+        if (m_worker.get() != nullptr)
+        {
+            l_jsonObj = m_worker->getSysCfgJsonObj();
+        }
+
+        std::error_code ec;
+
+        // Check if given path is filesystem path
+        if (!std::filesystem::exists(i_fruPath, ec) && (ec))
+        {
+            throw std::runtime_error("Given file path " + i_fruPath +
+                                     " not found.");
+        }
+
+        auto l_vpdStartOffset = jsonUtility::getVPDOffset(
```

Comment:
Line 235 onward does not align with the design discussion and design document presented.
The decision was to expose another public API from parser class to get the instance. Manager(or say any other external class) should only interact with parser class not factory.

Reply:
So ideally, a public API in parser class getVpdParserInstance() needs to be exposed which will be called here and it will do all the job from line 235 onward and then directly read API can be called on that instance.

Reply:
made the required changes